app/controllers/api/api_controller.rb

```diff
@@ -0,0 +1,146 @@
+class Api::ApiController < ApplicationController
+
+  respond_to :json
+
+  before_filter :set_params, only: [:single, :lca, :pept2pro]
+  before_filter :set_query, only: [:single, :lca]
+
+
+  def set_params
+    @sequences = params[:sequences]
+    if @sequences.kind_of? Hash
+      @sequences = @sequences.values
+    end
+    @sequences.map! &:chomp
+    @input_order = @sequences.dup
+
+    @equate_il = (!params[:equate_il].blank? && params[:equate_il] == 'true')
+    @full_lineage = (!params[:full_lineage].blank? && params[:full_lineage] == 'true')
+    @names = (!params[:names].blank? && params[:names] == 'true')
+
+    @sequences = @sequences.map {|s| s.gsub(/I/,'L') } if @equate_il
+  end
+
+  def set_query
+    if @full_lineage
+      if @names
+        @query = Taxon.includes(lineage: Lineage::ORDER_T)
+      else
+        @query = Taxon.includes(:lineage)
+      end
+    else
+      @query = Taxon
+    end
+
+  end
+
+  def single
+    @sequences = Sequence.joins(:peptides => :uniprot_entry).
```

Comment:
The join is actually needed, since where taking the taxon_id's from the uniprot_entries.


_[Original comment](https://github.ugent.be/unipept/unipept/pull/212#discussion_r1409) by an unknown user on Fri May 09 2014 at 14:52._